### PR TITLE
Refactoring portallocator away from a global var

### PR DESCRIPTION
**Pull request by paulbellamy**
Starts towards cleaning up #11626

### CODE REVIEW

**Comment by LK4D4** [comment]
@paulbellamy So, if this only start, can you describe your next steps, so we can discuss design.

File: daemon/networkdriver/portallocator/portallocator.go [hunk 1]
```diff
@@ -50,11 +50,20 @@ var (
 )
 
 var (
+	defaultIP            = net.ParseIP("0.0.0.0")
+	defaultPortAllocator = &PortAllocator{}
```
Comment (LK4D4) [inline daemon/networkdriver/portallocator/portallocator.go]: I think you can write `New` right here instead of `init()`

**Comment by paulbellamy** [comment]
The portallocator package now has a New(), as well as a default (similar to net/http). At the moment everywhere which is using it is still using the global default. So the next step seems to be to refactor the call-sites to use a specific PortAllocator.

Call sites are in: api/server, daemon, and daemon/networkdriver/portmapper. But, I'm not sure the best place to instantiate the PortAllocator, or pass it around. Adding it as an argument everywhere it's used seems quite messy.

File: daemon/networkdriver/portallocator/portallocator.go (hunk 1 above)
Comment (paulbellamy) [inline daemon/networkdriver/portallocator/portallocator.go]: Ah, yeah.

**Comment by LK4D4** [comment]
@paulbellamy It can be global variable(lol) in `daemon/networkdrvier/bridge/driver.go`. Which leads us to another problem that `networkdriver` is bunch of global variables too :) But at least we'll have pure from that ipallocator.

**Comment by LK4D4** [comment]
Cool. Thank you, I'll merge if tests will be okay.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**